### DISCUSSION TITLE
Looping over a subset of times in the `FieldSet`

### DISCUSSION THREAD

**Discussion by gmacgilchrist** [Parcels-code/Parcels#1318] [Q&A]
I wish to evolve particles in a velocity field that loops multiple times over a specific year (for example, a year with certain anomalous conditions). I thought I would be able to do this by setting `time_periodic` to `timedelta(365)` and setting the initialization time to be the start of the year that I wish to loop. However, this fails with:
```
ValueError: Time series provided is longer than the time_periodic parameter
```
The implication is that I can only *loop* when I wish to do so *across* the time boundaries of the FieldSet.

Is there a specific reason for this constraint? Would it be possible to remove it, and thus have the potential to loop over a subset of the `FieldSet`?

I am running with `parcels` version `2.3.0`.

**Comment by erikvansebille** [comment]
Thanks for bringing this up, @gmacgilchrist. I agree that this is not very clear in the documentation. 

The design philosophy of the `time_periodic` functionality is that the looping will be over the _full_ time-range of the `FieldSet`. If a user wants to run over a subset in time, they will have to make sure that the `FieldSet` is shortened (e.g., by providing fewer input netcdf files). 

The reason why Parcels still supports a `timedelta` object in `time_periodic` is that Parcels doesn't know what the time interval is between the last snapshot (t<sub>n</sub>) and the first snapshot (t<sub>0</sub>). Parcels could _assume_ that it's the same as between e.g. t<sub>0</sub> and t<sub>1</sub>, but we want to keep the flexibility to support `FieldSets` with variable time intervals.

So, this is not a bug but a design choice. I will transfer this Issue to a Discussion

**Comment by gmacgilchrist** [comment]
Thanks for the response @erikvansebille, this makes sense, and I agree with retaining flexibility.

I guess what I am wondering is whether it is possible to implement a feature of being able to loop _within the timeframe of one of the netcdfs provided to the `FieldSet`_. To offer my example, I have 6 years of data (2013 to 2018) for the BSOSE data assimilation. The data are arranged into single files containing all 6 years, with one file for each variable. I would like to loop over _just_ 2014 (a year with an anomalously strong Ross Gyre) and compare with trajectories looped over the full time-evolving field. To do this in the current set-up, I have to _extract_ the single year (2014) from the main netcdf file, save it in a new netcdf file, and use this as my `FieldSet`. Obviously this is not a problem in the grand scheme of things, but one could feasibly imagine someone wanting to do something similar for, say, every year of a 60+ year hindcast, where the data are saved in 5-year intervals, in which case they would have to resample all of their data to annual output.

I'm wondering if there is a reason why something like that _can't_ be performed in `parcels`?
